typescript/cli/src/deploy/warp.ts

```diff
@@ -627,17 +687,22 @@ async function updateExistingWarpRoute(
   );
   const transactions: AnnotatedEV5Transaction[] = [];
 
+  // Get all deployed router addresses
+  const deployedRoutersAddresses =
+    getRouterAddressesFromWarpCoreConfig(warpCoreConfig);
+
+  const expandedWarpDeployConfig = await expandWarpDeployConfig(
+    multiProvider,
+    warpDeployConfig,
+    deployedRoutersAddresses,
+  );
+
   await promiseObjAll(
-    objMap(warpDeployConfig, async (chain, config) => {
+    objMap(expandedWarpDeployConfig, async (chain, config) => {
       await retryAsync(async () => {
-        logGray(`Update existing warp route for chain ${chain}`);
-        const deployedConfig = warpCoreConfigByChain[chain];
-        if (!deployedConfig)
-          return logGray(
-            `Missing artifacts for ${chain}. Probably new deployment. Skipping update...`,
-          );
-
-        const deployedTokenRoute = deployedConfig.addressOrDenom!;
+        const deployedTokenRoute = deployedRoutersAddresses[chain];
+        assert(deployedTokenRoute, `Missing artifacts for ${chain}.`);
+
         const {
```

Comment:
This is outside the scope for this PR, but we really should be moving this into a constant IMO

Reply:
Created https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5543 and Assigned you @mshojaei-txfusion